app/scripts/lib/fxa-client.js

```diff
@@ -132,6 +132,8 @@ define(function (require, exports, module) {
       }
 
       var updatedSessionData = {
+        challengeMethod: accountData.challengeMethod,
```

Comment:
`confirmationMethod` doesn't seem like it fits with non-email challenge types like `sms` or `totp`.